cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -34,6 +34,133 @@ namespace modmesh
 namespace python
 {
 
+/// Execute the callback function with the typed array
+/// @tparam Callable the type of the callback function
+/// @param arrayplex the plex array, which is the wrapper of the typed array
+/// @param callback the callback function, which has the typed array as the argument
+/// @return the return type of the callback function
+template <typename Callable>
+// NOLINTNEXTLINE(misc-use-anonymous-namespace)
+static auto execute_callback_with_typed_array(SimpleArrayPlex & arrayplex, Callable && callback)
```

Comment:
getting the typed array from the array plex is a common usage, this design should save a lot of code.